### ISSUE TITLE
undefined method `underscore' for association class_name option

### ISSUE THREAD

**Issue by abhaynikam** [rails-api/active_model_serializers#2258]
#### Expected behavior vs actual behavior
It should not throw error if class_name is constantized while doing `underscore` on class_name or we should update the documentation if required.

#### Steps to reproduce
```
class Api::V1::Foo::UserController < BaseController

  def index
    respond_to do |format|
      format.json { render json: ..., each_serializer: Foo::User }
    end
  end
end
```

```
class Foo::User < ActiveModel::Serializer
  attribute :id, :name

  belongs_to :organization, class_name: Foo::Organization
end
```

https://github.com/rails-api/active_model_serializers/blob/0-10-stable/lib/active_model/serializer/reflection.rb#L59-L60

`undefined method 'underscore' for #<Class: >`

#### Environment

ActiveModelSerializers Version *(commit ref if not on tag)*: 0.10.7

Output of `ruby -e "puts RUBY_DESCRIPTION"`: 
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin17]

Integrated application and version *(e.g., Rails, Grape, etc)*: 
Rails version - 5.0.7

#### Backtrace
```
NoMethodError
undefined method `underscore' for #<Class:0x007f8bd5040e80>
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/activerecord-5.0.7/lib/active_record/dynamic_matchers.rb:21:in `method_missing'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/attr_encrypted-3.0.3/lib/attr_encrypted.rb:295:in `method_missing'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/attr_encrypted-3.0.3/lib/attr_encrypted/adapters/active_record.rb:129:in `method_missing_with_attr_encrypted'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/active_model_serializers-0.10.7/lib/active_model/serializer/reflection.rb:61:in `blockin initialize'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/active_model_serializers-0.10.7/lib/active_model/serializer/reflection.rb:58:in `fetch'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/active_model_serializers-0.10.7/lib/active_model/serializer/reflection.rb:58:in `initialize'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/active_model_serializers-0.10.7/lib/active_model/serializer.rb:246:in `new'
/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/active_model_serializers-0.10.7/lib/active_model/serializer.rb:246:in `belongs_to'
```

#### Additonal helpful information
```
# Gemfile.lock

active_model_serializers (0.10.7)
      actionpack (>= 4.1, < 6)
      activemodel (>= 4.1, < 6)
      case_transform (>= 0.2)
      jsonapi-renderer (>= 0.1.1.beta1, < 0.3)
```

**Comment by bf4** [comment]
Thanks for the report and due diligence!

Given that the option is called `class_name`, I don't think it's unreasonable to expect a string.  

Any reason you prefer a constant? Constants can create circular dependencies at boot.  

I'd accept a PR to clarify the documentation.

**Comment by abhaynikam** [comment]
@bf4 : Thanks for taking a look. No reason to prefer a constant. 

Just we were working on upgrading the rails application to `5.0.7`. So in process of upgrading Rails, we also upgraded the `active_model_serializers` version from `0.10.3` to `0.10.7`. This is when we started to get this error. 

I would love to raise a PR for documentation.

Also, I think it would be good if we raise an exception if the class_name was constant. But I agree it is expected to send class_name as a string instead of constant. So we can skip raising an exception you think that is not a valid request.
